tools/build_env/build_env.sh

```diff
@@ -40,18 +43,18 @@ if ! which docker > /dev/null; then
 fi
 
 if [[ $BUILD = true ]]; then
-  readonly DOCKER_ROOT="$(dirname $0)"
-  docker build --rm -t $IMAGE_NAME $DOCKER_ROOT
+  readonly DOCKER_ROOT="$(dirname $0)/$TAG"
```

Comment:
I don't think tags are intended to be used the way you are using. The idea is to use them for versioning (latest, rc, beta, next, 1.0, ...).

Instead, we should have a different image, with a different name. The indication is that you actually have a separate Dockerfile.